sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -285,10 +293,9 @@ Status getStatus() {
    * @return An unmodifiable view of the attributes associated wit this span
    */
   @VisibleForTesting
+  @GuardedBy("lock")
   Map<String, AttributeValue> getAttributes() {
-    synchronized (lock) {
-      return Collections.unmodifiableMap(attributes);
-    }
+    return Collections.unmodifiableMap(attributes);
```

Comment:
Do you need to wrap this in unmodifiable?

Reply:
I think we should keep it this way, otherwise tests might accidentally modify it.